spec-0008/index.md

```diff
@@ -0,0 +1,227 @@
+The Python Software Foundation (PSF) is also taking the importance of the OSSC seriously, as demonstrated by the [creation of the PSF Security Developer in Residence position in 2023](https://pyfound.blogspot.com/2023/06/announcing-our-new-security-developer.html).
+
+With the creation of the [Supply-chain Levels for Software Artifacts (SLSA) framework](https://slsa.dev/) and [OpenID Connect (OIDC)](https://openid.net/developers/how-connect-works/) standard being more widely adopted there are now high level developer tools, maintained by professional security teams, that are available with clear recommendations on how to use them.
+
+This SPEC outlines the requirements for practically adopting these security tools and recommendations with a focus on securely publishing release artifacts.
+Securely building release artifacts will be covered in a later SPEC. This set of recommendations complements the recommendations from [SPEC 6 — Keys to the Castle](https://github.com/scientific-python/specs/blob/main/spec-0006/index.md).
+
+This SPEC is written with GitHub in mind, note these recommendations still apply with other services such as GitLab and we refer project to these services' documentations and guidelines.
+
+## Implementation
+
+With a focus on securing the release artifact distribution process, the following processes and standards should be adopted.
+
+### Document the release process
+
+The release process should be clearly and fully documented in the developer documentation and describe each step to make a release and the permissions required to do so.
+
+### Hardening workflow environment permissions
+
+Workflows that publish release artifacts should have _run triggers_ that require intentional actions by maintainers (e.g., `workflow_dispatch` in GitHub Actions) and require multiple maintainers to approve the workflow to run (c.f. "Use GitHub Actions environments" section below).
+This is to safeguard the project from any one maintainer having the ability to commit to the default branch and make a release directly.
+
+It is also strongly recommended that the repository requires [signed commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) so that all releases have a verified commit to which they correspond.
+The branch from which the release is made should also be protected.
+
+#### Restrict permissions in CI runners to the minimum required
+
+To restrict the attack surface area of arbitrary code execution in CI runners, the default permissions the runners have should be restricted to the minimum possible (read access). In the GitHub Action workflow, this is accomplished by defining the following workflow global permissions block before any jobs are defined.
+
+```yaml
+permissions:
+  content: read
```

Comment:
```suggestion
  contents: read
```

Well this is embarrassing. [This should have been `contents`](https://github.com/scikit-hep/pyhf/blob/dff116cd5b3a4527501540a681813ad14d7a55fc/.github/workflows/ci.yml#L18-L19)! :grimacing: 

As @lagru said though, https://docs.github.com/en/rest/authentication/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-contents shows what all the GitHub operations that can touch repository "contents" are and what the required permissions are. The most restrictive is to just have `contents: read` and it is a good way to also ensure that you're not giving permissions to things unnecessarily.